src/main/java/com/openshift/jenkins/plugins/util/ClientCommandBuilder.java

```diff
@@ -45,7 +48,39 @@ public ClientCommandBuilder(String server, String project, boolean skipTLSVerify
             return n;
         }
         for (Object o : l) {
-            n.add(o.toString());
+            String s = o.toString();
+            if (s.contains(" ")) {
+                // the Jenkins QuotedStringTokenizer is not quite 
+                // copacetic with template parameters defined in 
+                // groovy variables (specifically strings where you have spaces but don't have 
+                // to encapsulate with " or ').
```

Comment:
I am not sure if I fully understand this. Can you list some examples?

Reply:
I'll see about adding some more specific examples in the comment.

And yes, limited use of `QuotedStringTokenizer`, or more machinations here, are the paths I'm considering to deal with the `rsh` / `exec` scenarios vs. the `ASDF=${asdf}` scenario during template parameter passing.

We technically do have a spec, albeit example driven, which just lacked precision wrt these groovy nuances with our API.  I don't see us investing in a formalized spec for this beyond the current help.jelly, example based approach.  Perhaps we start adding warning messages when we detect `ASDF=${asdf}` in the parameter passing that the value should be quoted, and that we'll start strictly enforcing that at a later date, but we'll have to live with whatever short term hack I land on for the time being.  We'll see ...